test/e2e/event_transformation_test.go

```diff
@@ -0,0 +1,151 @@
+// +build e2e
+
+/*
+Copyright 2019 The Knative Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/knative/eventing/pkg/apis/eventing/v1alpha1"
+	"github.com/knative/eventing/test"
+	pkgTest "github.com/knative/pkg/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+)
+
+/*
+TestEventTransformation tests the following scenario:
+
+EventSource ---> Channel ---> Subscriptions ---> Channel ---> Subscriptions ----> Service(Logger)
```

Comment:
This ASCII art doesn't quite match the comment on line 113, which essentially says that the transformation service is plugged inline between the two channels. The picture shows it as a T.

Reply:
Not sure if I got your point. 
If a Subscription specifies both Channel and Service, whenever it receives a new event, it will first forward the event to the Service and wait for it replying a transformed event, then it will send the transformed event to the Channel.

I've added more comments in the code.

Reply:
May be this?
```
             1              2                5                   6             7
EventSource ---> Channel ---> Subscription ---> Channel ---> Subscription ----> Service(Logger)
                                   |  ^
                                 3 |  | 4
                                   |  |
                                   |  ---------          
                                   -----------> Service(Transformation)
```

Reply:
Yeah this looks much clearer. Thanks for the suggestion! I'll change it.